plugins/woocommerce/tests/php/src/Internal/PushNotifications/PushNotificationsTest.php

```diff
@@ -175,6 +176,63 @@ public function test_it_caches_enablement_state_correctly() {
 		$this->assertTrue( $push_notifications_2->should_be_enabled(), 'Should return cached true value' );
 	}
 
+	/**
+	 * @testdox Tests that push_token post type is registered when enabled.
+	 */
+	public function test_it_registers_push_token_post_type_when_enabled() {
+		$this->set_up_jetpack_connection_manager_mock( array( 'is_connected' ) );
+
+		$this->jetpack_connection_manager_mock
+			->expects( $this->once() )
+			->method( 'is_connected' )
+			->willReturn( true );
+
+		$push_notifications = new PushNotifications();
+		$push_notifications->register();
+
+		$this->setExpectedIncorrectUsage( 'Automattic\WooCommerce\Blocks\Integrations\IntegrationRegistry::register' );
+		$this->setExpectedIncorrectUsage( 'WP_Block_Type_Registry::register' );
+
+		// phpcs:ignore WooCommerce.Commenting.CommentHooks.MissingHookComment
+		do_action( 'init' );
```

Comment:
Calling `do_action( 'init' )` as part of the test will probably trigger initialization of a lot more things than we want, which can have a lot of side effects.

I'd suggest we just check that `PushNotifications` has registered its `register_post_types()` method by using `has_action( 'init', ... )` and then we just call it to ensure the feature was enabled (or not) depending on the conditions.

(This also applies to the other instances of `do_action( 'init' )`).